guides/source/threading_and_code_execution.md

```diff
@@ -5,65 +5,145 @@ Threading and Code Execution in Rails
+(concurrently) in order for an application to run more efficiently. In this
+section, we will explore some of the ways this happens behind the scenes.
 
-When using a threaded web server, such as the default Puma, multiple HTTP
-requests will be served simultaneously, with each request provided its own
-controller instance.
+When using a threaded web server (such as Rails' default server, Puma) requests
+will be served simultaneously as each request is given its own controller
+instance.
 
-Threaded Active Job adapters, including the built-in Async, will likewise
-execute several jobs at the same time. Action Cable channels are managed this
-way too.
+Threaded Active Job adapters, including the built-in Async adapter, will
+likewise execute several jobs at the same time. Action Cable channels are
+managed this way too.
 
-These mechanisms all involve multiple threads, each managing work for a unique
-instance of some object (controller, job, channel), while sharing the global
-process space (such as classes and their configurations, and global variables).
-As long as your code doesn't modify any of those shared things, it can mostly
-ignore that other threads exist.
+Asynchronous Active Record queries are also performed in the background,
+allowing other processes to run on the main thread.
 
-The rest of this guide describes the mechanisms Rails uses to make it "mostly
-ignorable", and how extensions and applications with special needs can use them.
+The above mechanisms all involve multiple threads, each managing work for a
+unique instance of an object (controller, job, channel), while sharing the
+global process space (such as classes and their configurations, and global
+variables). As long as the code on each thread doesn't modify anything shared,
+multiple threads can safely run concurrently.
 
-Executor
---------
+Rails' built-in concurrency will cover the day-to-day needs of most application
+developers, and ensure applications remain generally performant.
 
-The Rails Executor separates application code from framework code: any time the
-framework invokes code you've written in your application, it will be wrapped by
-the Executor.
+NOTE: You can read more about how to configure Rails' concurrency in the
+[Framework Behavior](#framework-behavior) section.
 
-The Executor consists of two callbacks: `to_run` and `to_complete`. The Run
-callback is called before the application code, and the Complete callback is
-called after.
 
-### Default Callbacks
+### Isolated Execution State
 
-In a default Rails application, the Executor callbacks are used to:
+The `active_support.isolation_level` value in your `configuration.rb` file
+provides you the option to define where Rails' internal state should be stored
+while tasks are run. If you use a fiber-based server or job processor (e.g.
+[`falcon`](https://github.com/socketry/falcon)), you should set this value to
+`:fiber`, otherwise it is best to set it to `:thread`.
```

Comment:
We've historically avoided naming & linking specific external projects that aren't dependencies (because as soon as we mention one, we'll get follow-up PRs from progressively-more-obscure alternatives that want to be part of the list).

"`configuration.rb` file"?